### PR TITLE
options methods added to api

### LINKED ISSUE #117: Implement OPTIONS method in the POST API
This is the API that we will be working with for the rest of the quarter: #88

### PR DESCRIPTION
**Pull request by Abenezer1041**
close #117 

description
* Add OPTIONS method to pass pre-flight cors error.
* Option method is run before post request

Time:
| Task Breakdown Description      | Approx. Time |
| ----------- | ----------- |
| added the route to API and enabled cores on api | 2 hour       |
| Tested api  | 1 hours        |

Test
1. If you haven't clone the repository 
2. After cloning in the terminal type npm install
3. After installation is done type npm start. The app should start to run
4. open developer tools and click on network
5. Type a sentence and click on submit
Screen Shot 2022-03-14 at 10 49 17 PM
6. Verify that the request went through
Screen Shot 2022-03-14 at 11 04 00 PM

### CODE REVIEW

File: backend/simpleLambda/index.js [hunk 1]
```diff
@@ -51,22 +51,23 @@ exports.handler = async (event, context) => {
...
         
     } catch (err) {
         statusCode = 400;
         body = err.message;
-    } finally {
-        body = JSON.stringify(body)
     }
     
-    return body;
+    return {
```
Comment (dansjack) [inline backend/simpleLambda/index.js]: Only the `body` should be returned, not the `statusCode` or `headers`

File: backend/simpleLambda/index.js [hunk 2]
```diff
@@ -13,9 +13,9 @@ exports.handler = async (event, context) => {
         // break up sentences
         let feedback = event.feedback;
         let sentenceArray = [];
-        let sentenceObject = {};
         let newSentence = "";
         for (let i = 0; i < feedback.length; i++) {
+            let sentenceObject = {};
             newSentence = newSentence + feedback[i];
```
Comment (dale108) [inline backend/simpleLambda/index.js:20]: I think you probably want to initialize `newSentence` inside the loop, right?

File: backend/simpleLambda/index.js (hunk 1 above)
Comment (Abenezer1041) [reply backend/simpleLambda/index.js]: Thanks for the feedback. i changed it. it only return the body now.

**Review by dansjack — APPROVED** [review]

**Review by TurkiBK — APPROVED** [review]

File: backend/simpleLambda/index.js (hunk 2 above)
Comment (Abenezer1041) [reply backend/simpleLambda/index.js:20]: When i tried to put it inside the loop, i got a response with no sentence. I think it being outside the loop works better.
Screen Shot 2022-03-21 at 2 07 53 AM

**Review by mfern08 — COMMENTED** [review]
In this PR, I've noticed that  'https://d5z72uewg7.execute-api.us-west-2.amazonaws.com/feedback/kj-6exxg-20220222-lambda' is being used as the URL in the SubmitButton.js
Screen Shot 2022-03-21 at 9 32 57 AM

For my Sprint 5, I've implemented some changes in the SubmitButton.js, including endpoint url. It's now using the custom domain 'https://api.holodium.com/v1/feedback' for tomorrow's demo. The OPTIONS and POST works on the custom domain now as well. Yay! 

I think there's a merge conflict on your PR currently and wanted to inform you of these changes. It might be best to update the SubmitButton.js in your branch. I've also added a screenshot of how the SubmitButton.js looks now in development and my [PR](https://github.com/North-Seattle-College/ad440-winter2022-tuesday-repo/pull/164) for it. 

Screen Shot 2022-03-21 at 9 38 18 AM

**Comment by Abenezer1041** [comment]
Thanks Maria i got the latest change from development and fixed the conflict.

**Pull request merged**